### PR TITLE
[Core] Avoid incompatible variables in VTKoutput

### PR DESCRIPTION
**Pull request by loumalouomega**
Even if not added was considered in the FieldData definition

### CODE REVIEW

File: kratos/input_output/vtk_output.cpp [hunk 1]
```diff
@@ -384,6 +384,32 @@ void VtkOutput::WriteCellType(const TContainerType& rContainer, std::ofstream& r
...
+    } else if (KratosComponents<Variable<Vector>>::Has(rVariableName)){
+        return true;
+    } else if (KratosComponents<Variable<array_1d<double, 4>>>::Has(rVariableName)){
+        return true;
+    } else if (KratosComponents<Variable<array_1d<double, 6>>>::Has(rVariableName)){
+        return true;
+    } else if (KratosComponents<Variable<array_1d<double, 9>>>::Has(rVariableName)){
+        return true;
+    } else {
+        return false;
```
Comment (adityaghantasala) [inline kratos/input_output/vtk_output.cpp:406]: Shall we write a debug warning here ?

**Comment by adityaghantasala** [comment]
Ok for me. A comment : see above !

File: kratos/input_output/vtk_output.cpp [hunk 2]
```diff
@@ -467,9 +525,9 @@ void VtkOutput::WriteElementResultsToFile(const ModelPart& rModelPart, std::ofst
         }
 
         // Direct write GP values
-        for (IndexType entry = 0; entry < gauss_point_variables_extrapolated_to_nodes.size(); ++entry) {
-            const std::string& r_condition_result_name = gauss_point_variables_extrapolated_to_nodes[entry].GetString();
-            WriteGeometricalContainerIntegrationResults(r_condition_result_name,r_local_mesh.Elements(),rFileStream);
+        for (IndexType entry = 0; entry < gauss_point_variables_in_elements.size(); ++entry) {
```
Comment (adityaghantasala) [inline kratos/input_output/vtk_output.cpp:528]: good catch !

**Comment by loumalouomega** [comment]
Is already printed later (when actually reading the variables)

**Comment by adityaghantasala** [comment]
Ok now I see. Only concern it that such a check is done twice !! :-| 
May be we can avoid it in a future PR. 

Any way. This change is file for me.

**Review by adityaghantasala — APPROVED** [review]

**Comment by loumalouomega** [comment]
For doing that we should remove the evil variables, and it is a principle
not to modify Parameters.

**Pull request merged**